test/src/unittests/machinelearning/test_tensorflowpredict.py

```diff
@@ -59,19 +58,76 @@ def testRegression(self):
         pool = Pool()
         pool.set('model/Placeholder', batch)
 
-        tfp = TensorflowPredict(graphFilename=model,
-                                inputs=['model/Placeholder'],
-                                outputs=['model/Softmax'],
-                                isTraining=False,
-                                isTrainingName='model/Placeholder_1')
+        tfp = TensorflowPredict(**parameters)
         poolOut = tfp(pool)
 
         foundValues = poolOut['model/Softmax'].mean(axis=0).squeeze()
 
         self.assertAlmostEqualVector(foundValues, expectedValues, 1e-5)
 
-    def testInvalidFilename(self):
-        self.assertConfigureFails(TensorflowPredict(), {'graphFilename': ''})
+    def testRegressionFrozenModel(self):
+        parameters = {
+            'graphFilename': join(testdata.models_dir, 'vgg', 'vgg4.pb'),
+            'inputs': ['model/Placeholder'],
+            'outputs': ['model/Softmax'],
+            'isTraining': False,
+            'isTrainingName': 'model/Placeholder_1',
+        }
+
+        self.regression(parameters)
+
+    def testRegressionSavedModel(self):
+        parameters = {
+            'savedModel': join(testdata.models_dir, 'vgg', 'vgg4'),
+            'inputs': ['model/Placeholder'],
+            'outputs': ['model/Softmax'],
+            'isTraining': False,
+            'isTrainingName': 'model/Placeholder_1',
+        }
+
+        self.regression(parameters)
+
+    def testSavedModelPreferecce(self):
```

Comment:
"Preference". But a more informative test name is better. `testSavedModelOverridesGraphFilename` ?